doc/source/config.rst

```diff
@@ -0,0 +1,130 @@
+Configuration
+=============
+
+Satpy has two levels of configuration that allow to control how Satpy and
+its various components behave. There are a series of "settings" that change
+the global Satpy behavior. There are also a series of "component
+configuration" YAML files for controlling the complex functionality in readers,
+compositors, writers, and other Satpy components that can't be controlled
+with traditional keyword arguments.
+
+Settings
+--------
+
+There are configuration parameters in Satpy that are not specific to one
+component and control more global behavior of Satpy. These parameters can be
+set in one of three ways:
+
+1. Environment variable
+2. YAML file
+3. At runtime with ``satpy.config``
+
+This functionality is provided by the :doc:`donfig <donfig:configuration>`
+library. The currently available settings are described below.
+Each option is available from all three methods. If specified as an
+environment variable or specified in the YAML file on disk, it must be set
+**before** Satpy is imported.
+
+**YAML Configuration**
+
+YAML files that include these parameters can be in any of the following
+locations:
+
+1. ``<python environment prefix>/etc/satpy/satpy.yaml``
+2. ``~/.config/satpy/satpy.yaml``
+3. ``~/.satpy/satpy.yaml``
+4. ``<SATPY_CONFIG_PATH>/satpy.yaml`` (see :ref:`config_path_setting` below)
+
+**At runtime**
+
+After import, the values can be customized at runtime by doing:
+
+.. code-block:: python
+
+    import satpy
+    satpy.config.set(cache_dir="/my/new/cache/path")
+    # ... normal satpy code ...
+
+Or for specific blocks of code:
+
+.. code-block:: python
+
+    import satpy
+    with satpy.config.set(cache_dir="/my/new/cache/path"):
+        # ... some satpy code ...
+    # ... code using the original cache_dir
+
+cache_dir
+^^^^^^^^^
+
+**Environment variable**: SATPY_CACHE_DIR
+**Default**: See below
```

Comment:
How about having the section title as `Cache dir` and an extra line like
`**YAML and set name**`?

Reply:
Yeah I can try to do that. I might go even further and make it fully human-readable "Cache Directory". I might call it "Keyword Argument" or something.

Reply:
Done. Went with ``**YAML/Config Key**``.